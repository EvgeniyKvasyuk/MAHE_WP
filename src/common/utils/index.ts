export { noop } from './noop';
export { isTouchDevice } from './isTouchDevice';
