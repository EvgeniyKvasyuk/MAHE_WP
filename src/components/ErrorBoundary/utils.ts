export const handleError = (error: Error, info: { componentStack: string }) => {
  console.log(error, info, 'Message');
};
