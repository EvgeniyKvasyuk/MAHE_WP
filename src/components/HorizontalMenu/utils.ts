// TODO-Vladislav write tests on utils
import { TabProps } from './horizontal-menu-types';

export const getMenuId = (id: number) => `sub-menu-${id}`;

export const getTabById = (tabId: number, tabList: Array<TabProps>) => tabList.findIndex(({ id }) => id === tabId);

export const isElementInPopup = (element: HTMLElement, anchorEl: HTMLElement, id: number | null): boolean => {
  if (!id || !element) {
    return false;
  }
  const menuElement = document.getElementById(getMenuId(id));

  return !!menuElement?.contains?.(element);
};

const matchPath = (pathName: string, to: string): boolean => {
  const splittedPathName = pathName.split('/').filter(Boolean);

  return splittedPathName.includes(to.replace('/', ''));
};

export const getInitialTabId = (pathname: string, tabList: Array<TabProps>) => {
  const tab = tabList.find(({ to }) => matchPath(pathname, to));

  return tab?.id;
};
