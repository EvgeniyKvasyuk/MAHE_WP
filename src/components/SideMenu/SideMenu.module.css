@import '_sizes.css';
@import '_colors.css';
@import '_mixins.css';

.side-menu {
  &__paper {
    width: 24rem;

    @media (max-width: $screenMobileMax) {
      width: 100%;
    }
  }

  &__header {
    display: flex;
    align-items: center;
    padding: 1.5rem $app-bap-general-tablet-padding;

    @mixin header-padding;
  }

  &__close-icon {
    margin-right: 1rem;
    visibility: hidden;
    cursor: pointer;
    fill: $icon-color-primary;

    &--visible {
      visibility: visible;
    }

    &--right {
      margin-right: 0;
      margin-left: auto;
    }
  }

  &__logo {
    width: 10rem;
    height: 3.125rem;
  }

  &__body {
    padding: 1.5rem $app-bap-general-tablet-padding;

    @mixin header-padding;
  }
}
