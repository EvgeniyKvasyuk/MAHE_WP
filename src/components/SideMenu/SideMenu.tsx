import React, { ReactNode } from 'react';

import classNames from 'classnames/bind';

import { Drawer } from '@components/Drawer';
import { SvgIcon } from '@components/SvgIcon';

import styles from './SideMenu.module.css';

const cn = classNames.bind(styles);

export enum AnchorEnum {
  left = 'left',
  right = 'right',
}

export interface SideMenuProps {
  anchor?: AnchorEnum;
  children: ReactNode;
  onClose: VoidFunction;
  className?: string;
  open: boolean;
}

export function SideMenu({ anchor = AnchorEnum.left, children, onClose, open, className }: SideMenuProps) {
  return (
    <Drawer
      className={cn('side-menu', className)}
      paperClassName={cn('side-menu__paper')}
      anchor={anchor}
      open={open}
      onClose={onClose}
    >
      <div className={cn('side-menu__header')}>
        <SvgIcon
          icon="close"
          className={cn('side-menu__close-icon', {
            'side-menu__close-icon--visible': anchor === AnchorEnum.left,
          })}
          onClick={onClose}
        />

        <SvgIcon icon="mainLogo" className={cn('side-menu__logo')} viewBox="0 0 220 70" />

        <SvgIcon
          icon="close"
          className={cn('side-menu__close-icon', 'side-menu__close-icon--right', {
            'side-menu__close-icon--visible': anchor === AnchorEnum.right,
          })}
          onClick={onClose}
        />
      </div>

      <div className={cn('side-menu__body')}>{children}</div>
    </Drawer>
  );
}
