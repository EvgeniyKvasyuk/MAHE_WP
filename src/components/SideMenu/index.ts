export { SideMenu, AnchorEnum } from './SideMenu';
