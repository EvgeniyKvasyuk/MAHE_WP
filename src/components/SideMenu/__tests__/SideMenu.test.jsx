import React from 'react';

import { render } from '@testing-library/react';

import { noop } from '@common/utils';

import { SideMenu, AnchorEnum } from '../SideMenu';

test('<SideMenu> snapshot', () => {
  const menu = (
    <SideMenu open anchor={AnchorEnum.left} onClose={noop}>
      <div>test_menu_item1</div>
      <div>test_menu_item2</div>
    </SideMenu>
  );

  const component = render(menu);

  expect(component).toMatchSnapshot();
});
