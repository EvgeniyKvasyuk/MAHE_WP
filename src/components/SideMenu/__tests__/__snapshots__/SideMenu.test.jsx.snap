// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SideMenu> 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    style="padding-right: 1024px; overflow: hidden;"
  >
    <div
      aria-hidden="true"
    />
    <div
      class="MuiModal-root MuiDrawer-root MuiDrawer-modal side-menu css-1u2w381-MuiModal-root-MuiDrawer-root"
      role="presentation"
    >
      <div
        aria-hidden="true"
        class="MuiBackdrop-root css-i9fmh8-MuiBackdrop-root-MuiModal-backdrop"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      />
      <div
        data-test="sentinelStart"
        tabindex="0"
      />
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation16 MuiDrawer-paper MuiDrawer-paperAnchorLeft side-menu__paper css-4t3x6l-MuiPaper-root-MuiDrawer-paper"
        tabindex="-1"
      >
        <div
          class="side-menu__header"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium side-menu__close-icon side-menu__close-icon--visible css-i4bv87-MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <svg>
              close.svg
            </svg>
          </svg>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium side-menu__logo css-i4bv87-MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 220 70"
          >
            <svg>
              main_logo.svg
            </svg>
          </svg>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium side-menu__close-icon side-menu__close-icon--right css-i4bv87-MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <svg>
              close.svg
            </svg>
          </svg>
        </div>
        <div
          class="side-menu__body"
        >
          <div>
            test_menu_item1
          </div>
          <div>
            test_menu_item2
          </div>
        </div>
      </div>
      <div
        data-test="sentinelEnd"
        tabindex="0"
      />
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
  />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<SideMenu> snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    style="padding-right: 1024px; overflow: hidden;"
  >
    <div
      aria-hidden="true"
    />
    <div
      class="MuiModal-root MuiDrawer-root MuiDrawer-modal side-menu css-1u2w381-MuiModal-root-MuiDrawer-root"
      role="presentation"
    >
      <div
        aria-hidden="true"
        class="MuiBackdrop-root css-i9fmh8-MuiBackdrop-root-MuiModal-backdrop"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      />
      <div
        data-test="sentinelStart"
        tabindex="0"
      />
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation16 MuiDrawer-paper MuiDrawer-paperAnchorLeft side-menu__paper css-4t3x6l-MuiPaper-root-MuiDrawer-paper"
        tabindex="-1"
      >
        <div
          class="side-menu__header"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium side-menu__close-icon side-menu__close-icon--visible css-i4bv87-MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <svg>
              close.svg
            </svg>
          </svg>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium side-menu__logo css-i4bv87-MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 220 70"
          >
            <svg>
              main_logo.svg
            </svg>
          </svg>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium side-menu__close-icon side-menu__close-icon--right css-i4bv87-MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <svg>
              close.svg
            </svg>
          </svg>
        </div>
        <div
          class="side-menu__body"
        >
          <div>
            test_menu_item1
          </div>
          <div>
            test_menu_item2
          </div>
        </div>
      </div>
      <div
        data-test="sentinelEnd"
        tabindex="0"
      />
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
  />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
